vmm/src/vm.rs

```diff
@@ -775,16 +775,11 @@ impl Vm {
                     .add_fs(&mut _fs_cfg)
                     .map_err(Error::DeviceManager)?;
 
-                // Update VmConfig by adding the new device. This is important to
-                // ensure the device would be created in case of a reboot.
-                {
-                    let mut config = self.config.lock().unwrap();
-                    if let Some(fs_config) = config.fs.as_mut() {
-                        fs_config.push(_fs_cfg);
-                    } else {
-                        config.fs = Some(vec![_fs_cfg]);
-                    }
-                }
+                // Usually, the FsConfig would be added to the VmConfig so the
+                // device automatically gets recreated on reboot. Unfoutunately,
+                // virtiofsd exits when the fuse client disconnects, which means
+                // automatically recreating the device will cause the reboot to
+                // fail.
```

Comment:
The problem you're seeing with virtio-fs and reboot is not tied to the hotplug, it's simply not supported to reboot with a virtio-fs device attached.
That's why the problem needs to be fixed by having virtio-fs self spawning being supported at some point.
So I'd suggest that this PR should keep the initial code you wrote, as this is semantically correct.